discord/ext/commands/parameters.py

```diff
@@ -0,0 +1,269 @@
+    displayed_default: str = empty,
+) -> Any:
+    r"""parameter(\*, converter=..., default=..., displayed_default=...)
+
+    A way to assign custom metadata for a :class:`Command`\'s parameter.
+
+    Examples
+    --------
+
+    A custom converter, whilst annotating a parameter with a custom converter works at runtime, type checkers don't like it
+    cause they can't understand what's going on.
+
+    .. code-block:: python3
+
+        class SomeType:
+            foo: int
+
+        class MyVeryCoolConverter(commands.Converter[SomeType]):
+            ...  # implementation left as an exercise for the reader
+
+        @bot.command()
+        async def bar(ctx, cool_value: MyVeryCoolConverter):
+            cool_value.foo  # type checker warns MyVeryCoolConverter has no value foo (uh-oh)
+
+    However, fear not we can use :func:`parameter` to tell type checkers what's going on.
+
+    .. code-block:: python3
+
+        @bot.command()
+        async def bar(ctx, cool_value: SomeType = commands.parameter(converter=MyVeryCoolConverter)):
+            cool_value.foo  # no error (hurray)
+
+    A custom default can be used to have late binding behaviour
+
+    .. code-block:: python3
+
+        @bot.command()
+        async def wave(to: discord.User = commands.parameter(default=lambda ctx: ctx.author)):
+            await ctx.send(f'Hello {to.mention} :wave:')
+
+    Because this is such a common use-case, the library provides :obj:`.Author`, :obj:`.CurrentChannel` and
+    :obj:`.CurrentGuild`, armed with this we can simplify ``wave`` to:
+
+    .. code-block:: python3
+
+        @bot.command()
+        async def wave(to: discord.User = commands.Author):
+            await ctx.send(f'Hello {to.mention} :wave:')
+
+    :obj:`.Author` and co also have other benefits like having the displayed default being filled.
+
+    .. versionadded:: 2.0.0
+
+    Parameters
+    ----------
+    converter: Any
+        The converter to use for this parameter, this replaces the annotation at runtime which is transparent to type checkers.
+    default: Any
+        The default value for the parameter, if this is a :term:`callable` or a |coroutine_link|_ it is called with a
+        positional :class:`Context` argument.
+    displayed_default: :class:`str`
+        The displayed default in :attr:`Command.signature`.
+    """
+    return Parameter(
+        name="empty",
+        kind=inspect.Parameter.POSITIONAL_OR_KEYWORD,
+        annotation=converter,
+        default=default,
+        displayed_default=displayed_default,
+    )
+
+
+param = parameter
+r"""param(\*, converter=..., default=..., displayed_default=...)
+
+An alias for :func:`parameter`.
+"""
+
+# some handy defaults
+Author: Union[Member, User] = parameter(
+    default=attrgetter("author"),
+    displayed_default="<you>",
+    converter=Union[converter.MemberConverter, converter.UserConverter],
+)
+"""A default :class:`Parameter` which returns the :attr:`~.Context.author` for this context."""
```

Comment:
Can these floating docstrings be removed from the defaults since they're already explicitly enumerated in the API documentation?